user_funcs.py

```diff
@@ -118,5 +118,174 @@ def login(conn) -> tuple[str, int]:
+            if userid == followingid:
+                print("\nCan't follow self!")
+                return
+
+            # check if already following
+            curs.execute("SELECT * FROM \"following\" WHERE followerid = %s AND followingid = %s", (userid, followingid))
+            results = curs.fetchall()
+            if len(results) > 0:
+                print(f"\nAlready following {following_username}!")
+                # follow user if not already followed
+            else:
+                answer = input_utils.get_input_matching(f"follow {following_username}? y/n\n", regex="[yn]")
+                match answer:
+                    case "y":
+                        curs.execute("INSERT INTO \"following\" (followerid, followingid) VALUES (%s, %s)", (userid, followingid))
+                        print(f"\nNow following {following_username}!")
     conn.commit()
+
+
+def unfollow_user(conn, userid):
+    """
+    Guides user through unfollowing another user specified by email
+
+    :param conn: Connection to database
+    :param userid: ID of currently logged in user
+    """
+    with conn.cursor() as curs:
+        # prompt for email to unfollow
+        print("Who would you like to unfollow?")
+        email = input_utils.get_input_matching("Email: ", MAX_INPUT_LEN,  "^\S+@\S+\.\S+$", "Not a valid email address")
+
+        # check if user exists
+        curs.execute("SELECT userid, username FROM \"user\" WHERE email=%s", (email,))
+        results = curs.fetchall()
+        if len(results) == 0:
+            print(f"\nUser with email {email} doesn't exist!")
+        else:
+            followingid = results[0][0]
+            following_username = results[0][1]
+
+            # check if already following
+            curs.execute("SELECT * FROM \"following\" WHERE followerid = %s AND followingid = %s", (userid, followingid))
+            results = curs.fetchall()
+            # unfollow user
+            if len(results) > 0:
+                answer = input_utils.get_input_matching(f"unfollow {following_username}? y/n\n", regex="[yn]")
+                match answer:
+                    case "y":
+                        curs.execute("DELETE FROM \"following\" WHERE followerid = %s AND followingid = %s", (userid, followingid))
+                        print(f"\nUnfollowed {following_username}!")
+            else:
+                print(f"\nNot following {following_username}!")
+    conn.commit()
+
+
+def view_following(conn, userid):
+    """
+    Displays 10 other users that the current user follows at a time and gives 
+    the option to either return to following submenu, view more users, or unfollow a listed user
+
+    :param conn: Connection to database
+    :param userid: ID of currently logged in user
+    """
+    with conn.cursor() as curs:
+
+        action = ""
+        start_index = 0
+        end_index = 10
+        while action != "1":
+            # get next 10 users that are followed
```

Comment:
It's a little confusing to know what "page" I'm on. Can you output the start and end index (i.e. "Followed users 10-19:" )